### DISCUSSION TITLE
SImulations fail on the scoring server

### DISCUSSION THREAD

**Discussion by Sushant-Chavan** [hsr-project/general#7] [Q&A] [answered]
Hi, 
I am a member of a team participating in the robocup-at-home-2021 challenge. 

We are facing an issue when running the simulations on the devrt server. The server seems to build the docker images fine. But after that when it tries to run the simulation, it returns an [error](https://github.com/Sushant-Chavan/robocup-at-home-2021-challenge-1/commit/1924c526d7de980b6926172ce6c46bd784ff1bfb#commitcomment-51787847). 

Upon logging into the devrt website and checking the build status, the following error is displayed:

 `(HTTP code 500) server error - OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: initialization error: nvml error: driver/library version mismatch: unknown `

Please let us know what could be done regarding this.

Thanks and regards,
Sushant

**Comment by yosuke** [comment, marked as answer]
We have experienced this problem a few days ago due to the automatic update of the Nvidia driver on the simulation server.

I think the problem is fixed now. But, please let us know if you experience any further problems.

**Reply by Sushant-Chavan** [reply]
The simulations are working now. Thanks!

**Comment by yosuke** [comment]
@TomohiroOno33 Sorry we have experienced the system outage again (noticed you are suffered from the log).
Should be fixed very soon.

**Reply by TomohiroOno33** [reply]
Thank you for the announcement.
I'll test it using Worker.

**Reply by TomohiroOno33** [reply]
Sorry, I'm not sure if it's ok to ask here, but I have a question about starting the simulator.
If I do not get the following warning when starting the simulator, I think there is a problem that the robot does not work.
This problem happens about once in several times.

> simulator_1  | [ WARN] [1624074225.298966502, 1.005000000]: The complete state of the robot is not yet known.  Missing wrist_ft_sensor_frame_joint, base_b_bumper_joint, base_f_bumper_joint, base_l_passive_wheel_x_frame_joint, base_l_passive_wheel_y_frame_joint, base_l_passive_wheel_z_joint, base_r_passive_wheel_x_frame_joint, base_r_passive_wheel_y_frame_joint, base_r_passive_wheel_z_joint

**Reply by yosuke** [reply]
The simulation server is up now. Please let me know if there are any problems.

For the `The complete state of the robot is not yet known.` warning, it seems all the joints listed here are the `fixed` or `free` joint which state is known as obvious or undefined.

Also corresponds to the list of joints defined here:
https://github.com/hsr-project/hsrb_moveit_config/blob/master/config/hsrb.srdf#L99-L111

May be the issue of the timing of the moveit startup and SRDF loading?
SRDF should be loaded before the moveit starts.

**Reply by yosuke** [reply]
@TomohiroOno33
Is this problem happen only in your local simulator environment? or also happen in the build system?

**Reply by TomohiroOno33** [reply]
This is maybe happening only local.
I have confirmed that this is happening on multiple local computers.

> May be the issue of the timing of the moveit startup and SRDF loading?
> SRDF should be loaded before the moveit starts.

The problem happens when I run hsrb_robocup_dspl_docker.
Does it mean that I press the simulator start button too early?

**Reply by yosuke** [reply]
> Does it mean that I press the simulator start button too early?

Yes, this is most likely the reason.

**Reply by TomohiroOno33** [reply]
I understand.
I'll try to wait a bit before starting it.

**Comment by tiagojesilva** [comment]
Hi, Sushant-Chavan, I'm getting this error message when entering sudo docker-compose up command
`simulator_1  |     controllERROR: cannot launch node of type [laser_scan_matcher/laser_scan_matcher_node]: laser_scan_matcher
` have you experienced this problem?
